python/ray/data/datasource/numpy_datasource.py

```diff
@@ -28,14 +28,29 @@ class NumpyDatasource(FileBasedDatasource):
     _COLUMN_NAME = "data"
     _FILE_EXTENSION = "npy"
 
-    def _read_file(self, f: "pyarrow.NativeFile", path: str, **reader_args):
+    def __init__(
+        self,
+        paths: Union[str, List[str]],
+        numpy_load_args: Optional[Dict[str, Any]] = None,
```

Comment:
There was a bug where `numpy_load_args` wasn't getting passed to the `Datasource`. I've fixed it in this PR

Reply:
Should we add a test for that?